src/components/visualizations/chart/colorFactory.ts

```diff
@@ -39,98 +46,184 @@ export const attributeChartSupportedTypes = [
+                color: getRgbString(colorConfig.color)
+            };
         });
     }
 }
 
+function getAttributeColorMapping(attribute: any, colorPalette: IColorPalette, colorMapping: IColorMap[]) {
+    const itemsCount = attribute.items.length;
+    let currentColorPaletteIndex = 0;
+    return range(itemsCount).map((itemIndex) => {
+        const itemId = attribute.items[itemIndex].attributeHeaderItem.name;
+        const mappedColor = getColorFromMapping(itemId, colorPalette, colorMapping);
+
+        const color = mappedColor ? mappedColor : colorPalette[currentColorPaletteIndex % colorPalette.length];
+        currentColorPaletteIndex++;
+
+        return {
+            name: itemId,
+            localIdentifier: itemId,
+            guid: color.guid,
+            color: getRgbString(color)
+        };
+    });
+}
+
 export class AttributeColorStrategy extends ColorStrategy {
     protected createPalette(
         colorPalette: IColorPalette,
+        colorMapping: IColorMap[],
         _measureGroup: MeasureGroupType,
         viewByAttribute: any,
         stackByAttribute: any,
         _afm: AFM.IAfm
-    ): HighChartColorPalette {
-        const itemsCount = stackByAttribute ? stackByAttribute.items.length : viewByAttribute.items.length;
-        return range(itemsCount).map(itemIndex => getRgbString(colorPalette[itemIndex % colorPalette.length]));
+    ): IColorFullMapItem[] {
+        const attribute = stackByAttribute ? stackByAttribute : viewByAttribute;
+        return getAttributeColorMapping(attribute, colorPalette, colorMapping);
     }
 }
 
-export class HeatMapColorStrategy extends ColorStrategy {
+export class HeatmapColorStrategy extends ColorStrategy {
     public getColorByIndex(index: number): string {
-        return this.palette[index % this.palette.length];
+        return this.palette[index % this.palette.length].color;
     }
 
     protected createPalette(
         colorPalette: IColorPalette,
-        _measureGroup: MeasureGroupType,
+        colorMapping: IColorMap[],
+        measureGroup: MeasureGroupType,
         _viewByAttribute: any,
         _stackByAttribute: any,
         _afm: AFM.IAfm
-    ): HighChartColorPalette {
+    ): IColorFullMapItem[] {
+        let mappedColor;
+        if (colorMapping) {
+            const measureId = measureGroup.items[0].measureHeaderItem.localIdentifier;
+            mappedColor = getColorFromMapping(measureId, colorPalette, colorMapping);
+            if (mappedColor) {
+                return this.getCustomHeatmapColorPalette(mappedColor).map(color => ({ color }));
+            }
+        }
+
         if (colorPalette && isCustomPalette(colorPalette) && colorPalette[0]) {
-            return this.getCustomHeatmapColorPalette(colorPalette[0]);
+            return this.getCustomHeatmapColorPalette(colorPalette[0]).map(color => ({ color }));
         }
-        return HEATMAP_BLUE_COLOR_PALETTE;
+
+        return HEATMAP_BLUE_COLOR_PALETTE.map((color) => {
```

Comment:
maybe we could keep the same styling as above, i.e. `.map(color => ({ color }));`